src/script/lua_api/l_mapgen.cpp

```diff
@@ -1746,6 +1746,84 @@ int ModApiMapgen::l_serialize_schematic(lua_State *L)
 	return 1;
 }
 
+// read_schematic(schematic, options={...})
+int ModApiMapgen::l_read_schematic(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	SchematicManager *schemmgr = getServer(L)->getEmergeManager()->schemmgr;
+
+	//// Read options
+	// 2 = always write yslice, 1 = only if not default value (255), 0 = do not insert yslice table
+	u32 write_yslice = getintfield_default(L, 2, "write_yslice_prob", 2);
+
+	//// Get schematic
+	bool was_loaded = false;
+	Schematic *schem = (Schematic *)get_objdef(L, 1, schemmgr);
+	if (!schem) {
+		schem = load_schematic(L, 1, NULL, NULL);
+		was_loaded = true;
+	}
+	if (!schem) {
+		errorstream << "serialize_schematic: failed to get schematic" << std::endl;
+		return 0;
+	}
+	lua_pop(L, 2);
+
+	//// Create the Lua table
+	u32 numnodes = schem->size.X * schem->size.Y * schem->size.Z;
+	std::vector<std::string> names = schem->m_nodenames;
```

Comment:
const & reference. We don't need to copy this vector.

Reply:
Oh that's definitely right, sorry